packages/block-editor/src/components/link-control/use-search-handler.js

```diff
@@ -108,19 +109,22 @@ export default function useSearchHandler(
 	allowDirectEntry,
 	withCreateSuggestion
 ) {
-	const { fetchSearchSuggestions, pageOnFront, pageForPosts } = useSelect(
+	const { fetchSearchSuggestions, usePageSettings } = useSelect(
 		( select ) => {
 			const { getSettings } = select( blockEditorStore );
-
+			const settings = getSettings();
 			return {
-				pageOnFront: getSettings().pageOnFront,
-				pageForPosts: getSettings().pageForPosts,
+				usePageSettings: settings[ settingsKeys.usePageSettings ],
 				fetchSearchSuggestions:
-					getSettings().__experimentalFetchLinkSuggestions,
+					settings.__experimentalFetchLinkSuggestions,
 			};
 		},
 		[]
 	);
+	// The function should either be undefined or a stable function reference
+	// throughout the editor lifetime, much like importing a function from a
+	// module. Maybe warn if this becomes a common pattern and it does change?
+	const { pageOnFront, pageForPosts } = usePageSettings?.() ?? {};
```

Comment:
Is this the only place where we need these settings? It seems we only use them in a callback, so maybe a better settings would be a callback `getSiteSettings` or something like that no?

Reply:
Also something else, I've noted, it seems that the "link control" component relies on a lot of settings (`__experimentalFetchLinkSuggestions`) for instance and all these settings are not used elsewhere (unless I'm wrong). So maybe it's the way we're implementing the link control that is wrong, maybe we should just have a "default link control" and make it a filterable component like "mediaUpload". (Just thinking of alternatives, I'm not sure which is best)

Reply:
Yes, I think we should consider that too. I'm however not familiar enough with it to want to attempt a complete rewrite 😄

Reply:
I'm trying to make things better step by step and make sure are the dependencies are completely private so it's easier to rewrite in the future

Reply:
> It seems we only use them in a callback, so maybe a better settings would be a callback getSiteSettings or something like that no?

No, because it's used for rendered UI, not a callback. The selector is a resolver, so the component should re-render when the settings come in.

Reply:
It seems like `useSearchHandler` could entirely be a setting, but I'd prefer to attempt this separately because it would mean duplicating all this code to `edit-widgets`.

Reply:
> Also something else, I've noted, it seems that the "link control" component relies on a lot of settings (__experimentalFetchLinkSuggestions) for instance and all these settings are not used elsewhere (unless I'm wrong). So maybe it's the way we're implementing the link control that is wrong, 

@youknowriad I'd be grateful to hear more about how you'd like `LinkControl` to change. Now is the time because a lot of demands are being placed on it.

>...maybe we should just have a "default link control" and make it a filterable component like "mediaUpload". (Just thinking of alternatives, I'm not sure which is best)

I'd like to understand more about this.